sno/init.py

```diff
@@ -43,200 +57,368 @@ def import_gpkg(ctx, geopackage, table, list_tables):
+            dump_json_output({"sno.tables/v1": names}, sys.stdout)
         else:
-            click.secho(f"GeoPackage tables in '{Path(self.source).name}':", bold=True)
-            for table_name, identifier in table_list.items():
-                click.echo(f"{table_name}  -  {identifier}")
-        return table_list
+            click.secho(f"Tables found:", bold=True)
+            for table_name, pretty_name in names.items():
+                click.echo(f"  {table_name} - {pretty_name}")
+        return names
 
     def prompt_for_table(self, prompt):
-        table_list = self.print_table_list()
+        table_list = list(self.print_table_list().keys())
 
         if not sys.stdout.isatty():
             click.secho(
-                f'\n{prompt} via "{self.prefix}:{self.source}:MYTABLE"', fg="yellow",
+                f'\n{prompt} via `--table MYTABLE`', fg="yellow",
             )
             sys.exit(1)
 
-        t_choices = click.Choice(choices=table_list.keys())
-        t_default = next(iter(table_list)) if len(table_list) == 1 else None
+        t_choices = click.Choice(choices=table_list)
+        t_default = table_list[0] if len(table_list) == 1 else None
         return click.prompt(
             f"\n{prompt}", type=t_choices, show_choices=False, default=t_default,
         )
 
-    def __enter__(self):
+    def __str__(self):
+        s = str(self.source)
+        if self.table:
+            s += f":{self.table}"
+        return s
+
+    def check_table(self, prompt=False):
         if not self.table:
-            raise ValueError("No table specified")
+            if prompt:
+                # this re-inits and re-checks with the new table
+                self.table = self.prompt_for_table("Select a table to import")
+            else:
+                raise NotFound(
+                    "No table specified", exit_code=NO_TABLE, param_hint="--table"
+                )
+
+        if self.table not in self.get_tables():
+            raise NotFound(
+                f"Table '{self.table}' not found",
+                exit_code=NO_TABLE,
+                param_hint="--table",
+            )
+
+    def __enter__(self):
+        self.check_table()
 
-        dbcur = self.db.cursor()
-        dbcur.execute("BEGIN")
+        if self.ds.TestCapability(ogr.ODsCTransactions):
+            self.ds.StartTransaction()
         return self
 
     def __exit__(self, *exc):
-        self.db.cursor().execute("ROLLBACK")
+        if self.ds.TestCapability(ogr.ODsCTransactions):
+            self.ds.RollbackTransaction()
 
     @property
     @functools.lru_cache(maxsize=1)
     def row_count(self):
-        dbcur = self.db.cursor()
-        dbcur.execute(f"SELECT COUNT(*) FROM {gpkg.ident(self.table)};")
-        return dbcur.fetchone()[0]
+        # note: choose FAST method if possible ( i recall them being different with MITAB especially)
+        return self.ogrlayer.GetFeatureCount()
```

Comment:
API is documented as `GetFeatureCount(Layer self, int force=1) -> GIntBig`, so presumably we need `force=0`? Or there's the `OLCFastFeatureCount` capability